### PR TITLE
[DOCS] Actions, Checkpoint, ExpectationSuiteValidationResult, RunIdentifier related docstrings

### PR DESCRIPTION
**Pull request by joshua-stauffer**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], or [CONTRIB]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- Add doc strings

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in GitHub issues or Slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/checkpoint/actions.py [hunk 1]
```diff
@@ -76,15 +78,21 @@ def run(
...
-        :param validation_result_suite:
-        :param validation_result_suite_identifier:
-        :param data_asset:
-        :param expectation_suite_identifier:  The ExpectationSuiteIdentifier to use
-        :param checkpoint_identifier:  The Checkpoint to use
-        :param: kwargs - any additional arguments the child might use
-        :return:
+        Args:
+            validation_result_suite: An instance of the ExpectationSuiteValidationResult class.
+            validation_result_suite_identifier: an instance of either the ValidationResultIdentifier class (for open source Great Expectations) or the GeCloudIdentifier (from Great Expectations Cloud).
```
Comment (anthonyburdi) [inline great_expectations/checkpoint/actions.py]:
```suggestion
            validation_result_suite_identifier: an instance of either the ValidationResultIdentifier class (for open source Great Expectations) or the GXCloudIdentifier (from Great Expectations Cloud).
```

File: great_expectations/checkpoint/types/checkpoint_result.py [hunk 2]
```diff
@@ -156,14 +167,39 @@ def list_expectation_suite_names(self) -> List[str]:
...
+        Args:
+            group_by: Specify how the ExpectationValidationResults should be grouped.
+                Valid options are "validation_result_identifier", "expectation_suite_name",
+                "data_asset_name", or the default None. Providing an invalid group_by
+                value will cause this method to silently fail, and return None.
+
+        Returns:
+            A list of ExpectationSuiteValidationResult, when group_by=None
+            A dict of ValidationResultIdentifier keys and ExpectationValidationResults
+                values, when group_by="validation_result_identifier
```
Comment (anthonyburdi) [inline great_expectations/checkpoint/types/checkpoint_result.py]:
```suggestion
                values, when group_by="validation_result_identifier"
```

File: great_expectations/core/expectation_validation_result.py [hunk 3]
```diff
@@ -433,14 +433,72 @@ class ExpectationSuiteValidationResultMeta(TypedDict):
...
+          "data_asset_name": "taxi_data_1.csv",
+          "data_connector_name": "default_inferred_data_connector_name",
+          "datasource_name": "pandas"
+        },
+        "batch_markers": {
+          "ge_load_time": "20220727T154327.630107Z",
+          "pandas_data_fingerprint": "c4f929e6d4fab001fedc9e075bf4b612"
+        },
+        "batch_spec": {
+          "path": "/Users/josh/work/gx_example_projects/great_expectations/../data/taxi_data_1.csv"
```
Comment (anthonyburdi) [inline great_expectations/core/expectation_validation_result.py]:
```suggestion
          "path": "/Users/username/work/gx_example_projects/great_expectations/../data/taxi_data_1.csv"
```

**Review by anthonyburdi — APPROVED** [review]
Looks good with the suggestions noted!

File: great_expectations/core/expectation_validation_result.py (hunk 3 above)
Comment (joshua-stauffer) [reply great_expectations/core/expectation_validation_result.py]: lol nice catch - thanks

**Pull request merged**